# Desafio CIS-IEEE


Importar dependências e ler arquivo de entrada.


In [8]:
import pandas as pd

In [20]:
df_drugs = pd.read_csv('./Drugs.csv')

In [55]:
df_drugs.head(5)

ID    Age Gender                          Education Country Ethnicity  \
0   2  25-34      M                   Doctorate degree      UK     White   
1   3  35-44      M  Professional certificate/ diploma      UK     White   
2   4  18-24      F                     Masters degree      UK     White   
3   5  35-44      F                   Doctorate degree      UK     White   
4   6    65+      F            Left school at 18 years  Canada     White   

    Nscore   Escore   Oscore   AScore  ...  Ecstasy Heroin  Ketamine Legalh  \
0 -0.67825  1.93886  1.43533  0.76096  ...      CL4    CL0       CL2    CL0   
1 -0.46725  0.80523 -0.84732 -1.62090  ...      CL0    CL0       CL0    CL0   
2 -0.14882 -0.80615 -0.01928  0.59042  ...      CL0    CL0       CL2    CL0   
3  0.73545 -1.63340 -0.45174 -0.30172  ...      CL1    CL0       CL0    CL1   
4 -0.67825 -0.30033 -1.55521  2.03972  ...      CL0    CL0       CL0    CL0   

   LSD Meth Mushrooms Nicotine Semer  VSA  
0  CL2  CL3       CL0      CL4   CL0  CL0  
1  CL0  CL0       CL1      CL0   CL0  CL0  
2  CL0  CL0       CL0      CL2   CL0  CL0  
3  CL0  CL0       CL2      CL2   CL0  CL0  
4  CL0  CL0       CL0      CL6   CL0  CL0  

[5 rows x 32 columns]

Importações feitas e dataframe definido

### Respostas

1. No dataset existem alguns valores faltantes. Antes de começar a manipular os dados, trate essas informações e descreva sucintamente as alterações feitas.

In [ ]:
df_drugs.info()

In [ ]:
df_drugs.isna().sum()

É possível identificar que existem valores faltantes no dataset. O caso mais crítico é a coluna "Income (USD)". Primeiro trocarei o nome da coluna para facilitar a manipulação dos dados.

In [40]:
# Altera o nome da coluna para Income
df_drugs.rename(columns={'Income (USD)': 'Income'},inplace='True')

Vamos analisar os valores faltantes nas colunas e identificar os casos mais críticos.

In [41]:
# Quantas linhas tem pelo menos uma coluna com valor ausente
df_drugs[df_drugs.isna().any(axis=1)]

ID    Age Gender                                          Education  \
0        2  25-34      M                                   Doctorate degree   
1        3  35-44      M                  Professional certificate/ diploma   
2        4  18-24      F                                     Masters degree   
3        5  35-44      F                                   Doctorate degree   
4        6    65+      F                            Left school at 18 years   
...    ...    ...    ...                                                ...   
1877  1882  18-24      M                                  University degree   
1878  1883  18-24      M  Some college or university, no certificate or ...   
1879  1884  18-24      F  Some college or university, no certificate or ...   
1880  1885  18-24      M  Some college or university, no certificate or ...   
1881  1886  25-34    NaN                                  University degree   

     Country Ethnicity  Income   Nscore   Escore   Oscore  ...  Ecstasy  \
0         UK     White     NaN -0.67825  1.93886  1.43533  ...      CL4   
1         UK     White     NaN -0.46725  0.80523 -0.84732  ...      CL0   
2         UK     White     NaN -0.14882 -0.80615 -0.01928  ...      CL0   
3         UK     White     NaN  0.73545 -1.63340 -0.45174  ...      CL1   
4     Canada     White     NaN -0.67825 -0.30033 -1.55521  ...      CL0   
...      ...       ...     ...      ...      ...      ...  ...      ...   
1877  Canada     White     NaN -0.05188 -1.76250  0.88309  ...      CL0   
1878   Other       NaN     NaN -0.79151  0.32197  0.29338  ...      CL5   
1879     USA     White     NaN -1.19430  1.74091  1.88511  ...      CL0   
1880     USA     White     NaN -0.24649  1.74091  0.58331  ...      CL2   
1881     USA     White  1000.0  1.13281 -1.37639 -1.27553  ...      CL4   

      Heroin Ketamine  Legalh  LSD Meth Mushrooms Nicotine Semer  VSA  
0        CL0      CL2     CL0  CL2  CL3       CL0      CL4   CL0  CL0  
1        CL0      CL0     CL0  CL0  CL0       CL1      CL0   CL0  CL0  
2        CL0      CL2     CL0  CL0  CL0       CL0      CL2   CL0  CL0  
3        CL0      CL0     CL1  CL0  CL0       CL2      CL2   CL0  CL0  
4        CL0      CL0     CL0  CL0  CL0       CL0      CL6   CL0  CL0  
...      ...      ...     ...  ...  ...       ...      ...   ...  ...  
1877     CL0      CL0     CL2  CL0  CL0       CL2      CL2   CL0  CL0  
1878     CL2      CL0     CL4  CL5  CL4       CL0      CL6   CL0  CL1  
1879     CL0      CL0     CL3  CL3  CL0       CL0      CL0   CL0  CL5  
1880     CL0      CL0     CL3  CL5  CL4       CL4      CL5   CL0  CL0  
1881     CL0      CL2     CL0  CL2  CL0       CL2      CL6   CL0  CL0  

[1862 rows x 33 columns]

In [ ]:
# Mostra a quantidade de valores faltantes na coluna Income (crítico)
df_drugs.Income.isna().sum()

Existem duas possíveis abordagens para tratar os valores faltantes:
- Preencher os valores faltantes com a média dos valores existentes
- Remover linhas que possuam valores faltantes.
- Excluir os valores ausentes e desconsiderar a coluna

Para as colunas com poucos valores ausentes (<10) optarei por remover as poucas linhas com valores faltantes. Para as colunas com muitos valores ausentes (>10) optarei por excluir a coluna.

Primeiro vou tratar a coluna Income que possui muitos valores faltantes. Nesse caso, devido a quantidade alta de valores faltantes (proxima da quantidade total de linhas) é interessante desconsiderar a coluna.

In [48]:
df_drugs.drop(columns=['Income'],axis=1, inplace=True)

Agora para os outros valores ausentes que aparecem em poucos casos, podemos ou remover as linhas ou preencher com a média dos valores existentes. Vou optar por remover as linhas, visto que são muitas colunas que possuem valores faltantes.

In [54]:
df_drugs.dropna(axis=0, inplace=True)

Como passamos de 1862 linhas para 1853 linhas (redução de 0.4%), é possivel considerar a remoção das linhas com valores faltantes uma abordagem válida. Em casos onde muitas linhas foram removidas, seria interessante substituir os valores pela média.